broadcast-message.php

```diff
@@ -47,23 +52,38 @@
    }
    else
    {
+      $wanted_domains = array_intersect($allowed_domains, $_POST['domains']);
+
       $table_mailbox = table_by_key('mailbox');
-	  $table_alias = table_by_key('alias');
-      
-	  $q = "select username from $table_mailbox union select goto from $table_alias " .
-		   "where goto not in (select username from $table_mailbox)";
-
-      $result = db_query ($q);
-      if ($result['rows'] > 0)
-      {
+      $table_alias = table_by_key('alias');
+
+      $recipients = [];
+
+      foreach( $wanted_domains as $domain) {
+         $q = "SELECT username from $table_mailbox WHERE domain = '$domain'";
+         if(!isset($_POST['mailboxes_only']) || $_POST['mailboxes_only'] != "on") {
+            $q .= " UNION SELECT GOTO FROM $table_alias WHERE domain = '$domain' " .
+                  " AND goto NOT IN ($q)";
```

Comment:
Sorry for the delay, I'm too busy nowadays :-(

Instead of running multiple queries in a loop, please use an `IN` clause to match all requested domains in one query - something like `WHERE domain IN ('foo', 'bar')`

To make your life easier - you can use `db_in_clause()` to generate this part of the query ;-)

On the long term, using MailboxHandler and AliasHandler would be even better - but you don't need to do this unless you really want ;-)